### ISSUE TITLE
DataStore multiAuth is not working as expected

### ISSUE THREAD

**Issue by sadaqatdev** [aws-amplify/amplify-flutter#1693]
### Description

without login public models are not syn in android , iOS works fine 

```
Starting processing subscription events.
W/amplify:aws-api(22601): A subscription error occurred.
W/amplify:aws-api(22601): ApiAuthException{message=You must be signed-in with Cognito Userpools to be able to use getTokens, cause=null, recoverySuggestion=Sorry, we don't have a suggested fix for this error yet.}
W/amplify:aws-api(22601): 	at com.amplifyframework.api.aws.sigv4.DefaultCognitoUserPoolsAuthProvider.fetchToken(DefaultCognitoUserPoolsAuthProvider.java:98)
W/amplify:aws-api(22601): 	at com.amplifyframework.api.aws.sigv4.DefaultCognitoUserPoolsAuthProvider.getLatestAuthToken(DefaultCognitoUserPoolsAuthProvider.java:104)
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [X] API (GraphQL)
- [X] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Storage

### Steps to Reproduce

create public model and use iam auth and add data exception occur 

### Platforms

- [ ] iOS
- [X] Android

### Environment

```bash
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.4, on macOS 12.1 21C52 darwin-arm, locale en-PK)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
[✓] Chrome - develop for the web
[✓] Android Studio
[✓] VS Code (version 1.67.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
```

### Dependencies

```bash
Dart SDK 2.16.2
Flutter SDK 2.10.4
sister_help 1.0.0+1

dependencies:
- amplify_api 0.5.1 [amplify_api_plugin_interface amplify_core amplify_api_android amplify_api_ios collection flutter meta plugin_platform_interface]
- amplify_auth_cognito 0.5.1 [amplify_auth_cognito_android amplify_auth_cognito_ios amplify_auth_plugin_interface amplify_core collection flutter plugin_platform_interface]
- amplify_datastore 0.5.1 [flutter amplify_datastore_plugin_interface amplify_core plugin_platform_interface meta collection async]
- amplify_flutter 0.5.1 [amplify_analytics_plugin_interface amplify_api_plugin_interface amplify_auth_plugin_interface amplify_core amplify_datastore_plugin_interface amplify_flutter_android amplify_flutter_ios amplify_storage_plugin_interface collection flutter json_annotation meta plugin_platform_interface]
- connectivity_plus 1.1.0 [flutter meta connectivity_plus_platform_interface connectivity_plus_linux connectivity_plus_macos connectivity_plus_web connectivity_plus_windows]
- cupertino_icons 1.0.4
- cupertino_tabbar 2.0.0 [flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta typed_data vector_math sky_engine]
- flutter_dialogs 2.1.1 [flutter]
- flutter_image_compress 1.1.0 [flutter]
- flutter_launcher_icons 0.9.2 [args image path yaml]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path typed_data vector_math]
- hive 2.2.1 [meta crypto]
- hive_flutter 1.1.0 [flutter hive path_provider path]
- image_picker 0.8.5+3 [flutter image_picker_android image_picker_for_web image_picker_ios image_picker_platform_interface]
- intl 0.17.0 [clock path]
- mqtt_client 9.6.8 [typed_data event_bus path crypto meta]
- package_info_plus 1.4.2 [flutter package_info_plus_platform_interface package_info_plus_linux package_info_plus_macos package_info_plus_windows package_info_plus_web]
- pdf 3.6.4 [archive barcode crypto image meta path_parsing vector_math xml]
- permission_handler 9.2.0 [flutter meta permission_handler_android permission_handler_apple permission_handler_windows permission_handler_platform_interface]
- pin_code_fields 7.4.0 [flutter]
- provider 5.0.0 [collection flutter nested]
- qr_code_scanner 0.6.1 [js flutter flutter_web_plugins]
- qr_flutter 4.0.0 [flutter qr]
- share_plus 4.0.4 [meta mime flutter share_plus_platform_interface share_plus_linux share_plus_macos share_plus_windows share_plus_web]

transitive dependencies:
- amplify_analytics_plugin_interface 0.5.1 [amplify_core flutter meta]
- amplify_api_android 0.5.1 [flutter]
- amplify_api_ios 0.5.1 [amplify_core flutter]
- amplify_api_plugin_interface 0.5.1 [amplify_core collection flutter json_annotation meta]
- amplify_auth_cognito_android 0.5.1 [flutter]
- amplify_auth_cognito_ios 0.5.1 [amplify_core flutter]
- amplify_auth_plugin_interface 0.5.1 [amplify_core flutter meta]
- amplify_core 0.5.1 [collection date_time_format flutter meta plugin_platform_interface uuid]
- amplify_datastore_plugin_interface 0.5.1 [flutter meta collection amplify_core]
- amplify_flutter_android 0.5.1 [flutter]
- amplify_flutter_ios 0.5.1 [amplify_core flutter]
- amplify_storage_plugin_interface 0.5.1 [flutter meta amplify_core]
- archive 3.3.0 [crypto path]
- args 2.3.1
- async 2.8.2 [collection meta]
- barcode 2.1.0 [meta qr]
- characters 1.2.0
- charcode 1.3.1
- clock 1.1.0
- collection 1.15.0
- connectivity_plus_linux 1.3.0 [flutter connectivity_plus_platform_interface meta nm]
- connectivity_plus_macos 1.2.2 [connectivity_plus_platform_interface flutter]
- connectivity_plus_platform_interface 1.2.0 [flutter meta plugin_platform_interface]
- connectivity_plus_web 1.2.0 [connectivity_plus_platform_interface flutter_web_plugins flutter]
- connectivity_plus_windows 1.2.0 [connectivity_plus_platform_interface flutter]
- cross_file 0.3.3+1 [js meta]
- crypto 3.0.2 [typed_data]
- date_time_format 2.0.1
- dbus 0.7.3 [args ffi meta xml]
- event_bus 2.0.0
- ffi 1.2.1
- file 6.1.2 [meta path]
- flutter_plugin_android_lifecycle 2.0.6 [flutter]
- flutter_web_plugins 0.0.0 [flutter js characters collection material_color_utilities meta typed_data vector_math]
- http 0.13.4 [async http_parser meta path]
- http_parser 4.0.1 [collection source_span string_scanner typed_data]
- image 3.1.3 [archive meta xml]
- image_picker_android 0.8.4+13 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface]
- image_picker_for_web 2.1.8 [flutter flutter_web_plugins image_picker_platform_interface]
- image_picker_ios 0.8.5+5 [flutter image_picker_platform_interface]
- image_picker_platform_interface 2.5.0 [cross_file flutter http plugin_platform_interface]
- js 0.6.3
- json_annotation 4.5.0 [meta]
- material_color_utilities 0.1.3
- meta 1.7.0
- mime 1.0.2
- nested 1.0.0 [flutter]
- nm 0.5.0 [dbus]
- package_info_plus_linux 1.0.5 [package_info_plus_platform_interface flutter path]
- package_info_plus_macos 1.3.0 [flutter]
- package_info_plus_platform_interface 1.0.2 [flutter meta plugin_platform_interface]
- package_info_plus_web 1.0.5 [flutter flutter_web_plugins http meta package_info_plus_platform_interface]
- package_info_plus_windows 1.0.5 [package_info_plus_platform_interface ffi flutter win32]
- path 1.8.0
- path_parsing 0.2.1 [vector_math meta]
- path_provider 2.0.10 [flutter path_provider_android path_provider_ios path_provider_linux path_provider_macos path_provider_platform_interface path_provider_windows]
- path_provider_android 2.0.14 [flutter path_provider_platform_interface]
- path_provider_ios 2.0.9 [flutter path_provider_platform_interface]
- path_provider_linux 2.1.7 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_macos 2.0.6 [flutter path_provider_platform_interface]
- path_provider_platform_interface 2.0.4 [flutter platform plugin_platform_interface]
- path_provider_windows 2.0.7 [ffi flutter path path_provider_platform_interface win32]
- permission_handler_android 9.0.2+1 [flutter permission_handler_platform_interface]
- permission_handler_apple 9.0.4 [flutter permission_handler_platform_interface]
- permission_handler_platform_interface 3.7.0 [flutter meta plugin_platform_interface]
- permission_handler_windows 0.1.0 [flutter permission_handler_platform_interface]
- petitparser 4.4.0 [meta]
- platform 3.1.0
- plugin_platform_interface 2.1.2 [meta]
- process 4.2.4 [file path platform]
- qr 2.1.0 [meta]
- share_plus_linux 3.0.0 [share_plus_platform_interface file flutter meta url_launcher]
- share_plus_macos 3.0.0 [share_plus_platform_interface flutter]
- share_plus_platform_interface 3.0.2 [flutter meta mime plugin_platform_interface]
- share_plus_web 3.0.0 [share_plus_platform_interface url_launcher flutter flutter_web_plugins meta]
- share_plus_windows 3.0.0 [share_plus_platform_interface flutter meta url_launcher]
- sky_engine 0.0.99
- source_span 1.8.1 [collection path term_glyph]
- string_scanner 1.1.0 [charcode source_span]
- term_glyph 1.2.0
- typed_data 1.3.0 [collection]
- url_launcher 6.1.2 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- url_launcher_android 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.0.17 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.0.1 [flutter url_launcher_platform_interface]
- url_launcher_platform_interface 2.0.5 [flutter plugin_platform_interface]
- url_launcher_web 2.0.11 [flutter flutter_web_plugins url_launcher_platform_interface]
- url_launcher_windows 3.0.1 [flutter url_launcher_platform_interface]
- uuid 3.0.6 [crypto]
- vector_math 2.1.1
- win32 2.5.2 [ffi]
- xdg_directories 0.2.0+1 [meta path process]
- xml 5.3.1 [collection meta petitparser]
- yaml 3.1.1 [collection source_span string_scanner]
```

### Device

android Samsung G9

### OS

NA

### CLI Version

8.3.1

**Comment by Jordan-Nelson** [comment]
Hi @sadaqatdev - Can you please share your schema (or at least a subset of it with the public/private models)?

**Comment by sadaqatdev** [comment]
HI @Jordan-Nelson the following is  scheme

type RoomConnectServiceStatus @model @auth(rules: [{allow: private}, {allow: public, provider: iam, operations: [read]}]) {
  id: ID!
  isActivatedstatus: Boolean
  roomIdstatus: String
  serviceIDstatus: String
  currentDeviceUidstatus: String
  updateTimestatus: AWSDateTime
}

type MqttCloudConfig @model @auth(rules: [{allow: private}, {allow: public, provider: iam, operations: [read]}]) {
  id: ID!
  ip: String
  port: String
  password: String
  username: String
}

type Nurse @model @auth(rules: [{allow: private}]) {
  id: ID!
  name: String
  phone: AWSPhone!
  thumbnail: String
}

type Owner @model @auth(rules: [{allow: private}]) {
  id: ID!
  roomID: ID
  nurseID: ID
}

type Room @model @auth(rules: [{allow: private}, {allow: public, provider: iam, operations: [read]}]) {
  id: ID!
  title: String
}

type RoomConnecttoService @model @auth(rules: [{allow: private}, {allow: public, provider: iam, operations: [read]}]) {
  id: ID!
  roomID: ID
  serviceID: ID
  currentDeviceUid: String
  isActivated: Boolean
}

type Services @model @auth(rules: [{allow: private}, {allow: public, provider: iam, operations: [read]}]) {
  id: ID!
  name: String
  ordered: Boolean
  fulfilled: Boolean
  thumbnail: String
  roomConnectoId: String
}

type Patient @model @auth(rules: [{allow: private}]) {
  id: ID!
  name: String
  phone: AWSPhone!
  thumbnail: String
}

type SubscribedRooms @model @auth(rules: [{allow: private}]) {
  id: ID!
  roomID: ID
  userID: ID
}

**Comment by sadaqatdev** [comment]
yes I configure multi auth , but not working

**Comment by sadaqatdev** [comment]
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.sigv4.DefaultCognitoUserPoolsAuthProvider.fetchToken(DefaultCognitoUserPoolsAuthProvider.java:98)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.sigv4.DefaultCognitoUserPoolsAuthProvider.getLatestAuthToken(DefaultCognitoUserPoolsAuthProvider.java:104)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.SubscriptionAuthorizer.forCognitoUserPools(SubscriptionAuthorizer.java:147)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.SubscriptionAuthorizer.createHeaders(SubscriptionAuthorizer.java:96)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.SubscriptionAuthorizer.createHeadersForConnection(SubscriptionAuthorizer.java:71)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.SubscriptionEndpoint.buildConnectionRequestUrl(SubscriptionEndpoint.java:300)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.SubscriptionEndpoint.requestSubscription(SubscriptionEndpoint.java:127)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.MutiAuthSubscriptionOperation.dispatchRequest(MutiAuthSubscriptionOperation.java:113)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.MutiAuthSubscriptionOperation.lambda$iziEcYpvlINdYbit2it7fDbbt8A(Unknown Source:0)
W/amplify:aws-api(23071): 	at com.amplifyframework.api.aws.-$$Lambda$MutiAuthSubscriptionOperation$iziEcYpvlINdYbit2it7fDbbt8A.run(Unknown Source:2)
W/amplify:aws-api(23071): 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
W/amplify:aws-api(23071): 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
W/amplify:aws-api(23071): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
W/amplify:aws-api(23071): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:

**Comment by Jordan-Nelson** [comment]
Thanks @sadaqatdev. Can you also provide your amplifyconfig.json (with sensitive info masked or removed)?

**Comment by Jordan-Nelson** [comment]
I see you already commented on https://github.com/aws-amplify/amplify-flutter/issues/1220. This could be related, although it looks like that was resolved a while back.

**Comment by sadaqatdev** [comment]
const amplifyconfig = ''' {
    "UserAgent": "aws-amplify-cli/2.0",
    "Version": "1.0",
    "api": {
        "plugins": {
            "awsAPIPlugin": {
                "sisterhelpbg": {
                    "endpointType": "GraphQL",
                    "endpoint": " ",
                    "region": "eu-west-1",
                    "authorizationType": "AWS_IAM"
                }
            }
        }
    },
    "auth": {
        "plugins": {
            "awsCognitoAuthPlugin": {
                "UserAgent": "aws-amplify-cli/0.1.0",
                "Version": "0.1.0",
                "IdentityManager": {
                    "Default": {}
                },
                "AppSync": {
                    "Default": {
                        "ApiUrl": " ",
                        "Region": "eu-west-1",
                        "AuthMode": "AWS_IAM",
                        "ClientDatabasePrefix": " "
                    },
                    "sisterhelpbg_AMAZON_COGNITO_USER_POOLS": {
                        "ApiUrl": " ",
                        "Region": "eu-west-1",
                        "AuthMode": "AMAZON_COGNITO_USER_POOLS",
                        "ClientDatabasePrefix": " "
                    }
                },
                "CredentialsProvider": {
                    "CognitoIdentity": {
                        "Default": {
                            "PoolId": " ",
                            "Region": "eu-west-1"
                        }
                    }
                },
                "CognitoUserPool": {
                    "Default": {
                        "PoolId": " ",
                        "AppClientId": " ",
                        "Region": "eu-west-1"
                    }
                },
                "Auth": {
                    "Default": {
                        "authenticationFlowType": "USER_SRP_AUTH",
                        "socialProviders": [],
                        "usernameAttributes": [
                            "PHONE_NUMBER"
                        ],
                        "signupAttributes": [
                            "PHONE_NUMBER"
                        ],
                        "passwordProtectionSettings": {
                            "passwordPolicyMinLength": 8,
                            "passwordPolicyCharacters": []
                        },
                        "mfaConfiguration": "OFF",
                        "mfaTypes": [
                            "SMS"
                        ],
                        "verificationMechanisms": [
                            "EMAIL"
                        ]
                    }
                }
            }
        }
    }
}''';

**Comment by HuiSF** [comment]
From testing, I saw unexpected behaviors with DataStore with both amplify-swift and amplify-android libraries.

To summarize the issue:

When use mixed auth rules, i.e. public permission and private/owner permission, DataStore fails to allow the public access when there is no authenticated user session. 

Take schema 

```graphql
type UserProfile @model @auth(rules: [{ allow: owner }]) {
  id: ID!
  name: String!
}

type ModelA @model @auth(rules: [{ allow: public, provider: apiKey }]) {
  id: ID!
  content: String
}

type ModelB
  @model
  @auth(
    rules: [
      { allow: public, provider: apiKey, operations: [read] }
      { allow: owner }
    ]
  ) {
  id: ID!
  content: String
}

type ModelC
  @model
  @auth(
    rules: [
      { allow: public, provider: apiKey, operations: [read] }
      { allow: private, provider: userPools }
    ]
  ) {
  id: ID!
  content: String
}
```

When configure DataStore to use multi-auth mode with **NO authenticated session**:

### Expected Behavior on Android

As a developer, I want my end users to have `read` access to Model A B and C, including receiving subscription events.

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | Yes          | Yes           |
| ModelB      | Yes          | Yes           |
| ModelC      | Yes          | Yes           |

### Actual Behavior on Android

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | No           | No            |
| ModelB      | No           | No            |
| ModelC      | No           | No            |

**What happened:** with amplify-android, it attempted to create subscription for `UserProfile` , when there was no authenticated user session, the subscription failed, and put DataStore into the `LOCAL_ONLY` mode. The should-be-allowed read operations on other models were not working. This make the multi-auth mode unusable.

### Expected Behavior on iOS

As a developer, I want my end users to have `read` access to Model A B and C, including receiving subscription events.

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | Yes          | Yes           |
| ModelB      | Yes          | Yes           |
| ModelC      | Yes          | Yes           |

### Actual Behavior on iOS

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | Yes          | Yes           |
| ModelB      | Yes          | No            |
| ModelC      | Yes          | No            |

**What happened:** amplify-swift attempted to establish subscriptions for `ModelB` and `ModelC` using user credentials, after failure, it doesn’t attemp to establish subscriptions with API key. So the App cannot receive any update of `ModelB` and `ModelC` while the public read is allowed on these two models.

**Comment by HuiSF** [comment]
From testing, I saw unexpected behaviors with DataStore with both amplify-swift and amplify-android libraries.

**To summarize the issue:**

When use mixed auth rules, i.e. public permission and private/owner permission, DataStore fails to allow the public access when there is no authenticated user session. 

Take schema 

```graphql
type UserProfile @model @auth(rules: [{ allow: owner }]) {
  id: ID!
  name: String!
}

type ModelA @model @auth(rules: [{ allow: public, provider: apiKey }]) {
  id: ID!
  content: String
}

type ModelB
  @model
  @auth(
    rules: [
      { allow: public, provider: apiKey, operations: [read] }
      { allow: owner }
    ]
  ) {
  id: ID!
  content: String
}

type ModelC
  @model
  @auth(
    rules: [
      { allow: public, provider: apiKey, operations: [read] }
      { allow: private, provider: userPools }
    ]
  ) {
  id: ID!
  content: String
}
```

When configure DataStore to use multi-auth mode with **NO authenticated session**:

### Expected Behavior on Android

As a developer, I want my end users to have `read` access to Model A B and C, including receiving subscription events.

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | Yes          | Yes           |
| ModelB      | Yes          | Yes           |
| ModelC      | Yes          | Yes           |

### Actual Behavior on Android

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | No           | No            |
| ModelB      | No           | No            |
| ModelC      | No           | No            |

**What happened:** with amplify-android, it attempted to create subscription for `UserProfile` , when there was no authenticated user session, the subscription failed, and put DataStore into the `LOCAL_ONLY` mode. The should-be-allowed read operations on other models were not working. This make the multi-auth mode unusable.

### Expected Behavior on iOS

As a developer, I want my end users to have `read` access to Model A B and C, including receiving subscription events.

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | Yes          | Yes           |
| ModelB      | Yes          | Yes           |
| ModelC      | Yes          | Yes           |

### Actual Behavior on iOS

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | Yes          | Yes           |
| ModelB      | Yes          | No            |
| ModelC      | Yes          | No            |

**What happened:** amplify-swift attempted to establish subscriptions for `ModelB` and `ModelC` using user credentials, after failure, it doesn’t attemp to establish subscriptions with API key. So the App cannot receive any update of `ModelB` and `ModelC` while the public read is allowed on these two models.

I've created issues for both amplify-swift and amplify-android for further actions.

**Comment by MarlonJD** [comment]
I also quick tested with GraphQL API, cognito + API Key. Whenever I try to access without cognito user, even public model. It gives error.

**Comment by Jordan-Nelson** [comment]
This should be resolved in the next release (after https://github.com/aws-amplify/amplify-flutter/pull/3612 is merged). Given the following schema, below are the results for an unauthenticated user. The results are the same on iOS and Android.

| models      | sync queries | subscriptions |
| ----------- | ------------ | ------------- |
| UserProfile | No           | No            |
| ModelA      | Yes          | Yes           |
| ModelB      | Yes          | Yes           |
| ModelC      | Yes          | Yes           |


```graphql
type UserProfile @model @auth(rules: [{ allow: owner }]) {
  id: ID!
  name: String!
}

type ModelA @model @auth(rules: [{ allow: public, provider: apiKey }]) {
  id: ID!
  content: String
}

type ModelB
  @model
  @auth(
    rules: [
      { allow: public, provider: apiKey, operations: [read] }
      { allow: owner }
    ]
  ) {
  id: ID!
  content: String
}

type ModelC
  @model
  @auth(
    rules: [
      { allow: public, provider: apiKey, operations: [read] }
      { allow: private, provider: userPools }
    ]
  ) {
  id: ID!
  content: String
}
```

**Comment by MarlonJD** [comment]
Awesome, happy to hear that, I couldn't use datastore for this reasons. I'll try when it comes to release. Thanks!

**Comment by Jordan-Nelson** [comment]
As mentioned above, this should be resolved as of v1.4.0
